lib/active_model/serializer/adapter.rb

```diff
@@ -86,7 +87,11 @@ def meta_key
       end
 
       def root
-        serializer.json_key.to_sym if serializer.json_key
+        if serializer.json_key
+          serializer.json_key.to_sym
+        else
+          serializer.root
```

Comment:
I guess, I should have added a test for this, cause I ran in to a situation where json_key was nil. hmm. I'll look in to this